### PR TITLE
Allow to calculate multiple molecules directly ( clean update)

### LINKED ISSUE #63: Calculate multiple molecules directly
#  Problem

Current version of [calc_spectrum](https://radis.readthedocs.io/en/latest/source/radis.lbl.calc.html#radis.lbl.calc.calc_spectrum) is limited to one molecule only. 
We have to call it several times then use [MergeSlabs](https://radis.readthedocs.io/en/latest/los/los.html)

# Suggestion

- Allow calc_spectrum to take a dictionary as input for `mole_fraction=` 
- if a dictionary is given, then `calc_spectrum` calls himself several times, and returns a `MergeSlabs` directly 

(easy modification, great feature. Requires a test, an update to docstrings and examples in docs)

- [x] Implement a version version
- [ ] Choose syntax for user interface 
- [ ] Implement syntax

-------------

# Syntax for the Interface

Which syntax to implement in PR #74   ?

*(suggest your own if it seams more user-friendly / logical / prettier)*

### 1. Current version, where we could also allow default values

```
calc_spectrum(
    molecule={'CO2':"mole_fraction":0.8,  'CO':"mole_fraction":0.2}, 
    isotope=1,
    ... )
```
Would work. Then the default databank (fetch)  is used for everyone 

If default value check that it is the same for everyone :
```
calc_spectrum(
    molecule={'CO2':"isotope":1,  'CO':isotope=2}, 
    isotope=1, 
    ... )
```
Should fail 

### 2. Simpler syntax 

```
calc_spectrum(
    molecule={'CO2':0.8 ,  'CO':0.2 },
    ... )
```
Could work? 

Or 

### 3. Directly from the keys:

```
calc_spectrum(
    mole_fraction={'CO2':0.8 ,  'CO':0.2 }, 
    ... )
```
```
calc_spectrum(
    mole_fraction={'CO2': 0.8, 'CO':0.2}, 
    isotope={'CO2':[1,2], 'CO':[1,2,3]},
     ... )
```

### Vote!  (comment to add your own)

[](https://api.gh-polls.com/poll/01E11RWDWA19X63PP62S3GSKD0/Syntax%201/vote)
[](https://api.gh-polls.com/poll/01E11RWDWA19X63PP62S3GSKD0/Syntax%202/vote)
[](https://api.gh-polls.com/poll/01E11RWDWA19X63PP62S3GSKD0/Syntax%203/vote)

### PR DESCRIPTION
**Pull request by ulyssed**
### Description
Allow the use of calc_spectrum for multiple molecules spectra computations.
This is a clean update and PR following #74 .

Fixes #63

### CODE REVIEW

**Comment by erwanp** [comment]
Alright, tests passed, we're almost there! 

@ulyssed  could you update the documentation to show this new feature? 

For instance we could add two new paragraphs in [Calculating spectra](https://radis.readthedocs.io/en/latest/lbl/lbl.html#calculating-spectra).  

---

### Calculate one molecule

Here we calculate a CO spectrum at equilibrium, and plot the transmittance:
```
s = calc_spectrum(
        wavenum_min=1900,
        wavenum_max=2300,
        Tgas=700,
        path_length=0.1,
        molecule='CO',
        mole_fraction=0.5,
        isotope=1,
    )
s.plot('transmittance_noslit')
```

### Calculate multiple molecules

Here we add the contribution of CO2, and plot the transmittance:
```
s = calc_spectrum(
        wavenum_min=1900,
        wavenum_max=2300,
        Tgas=700,
        path_length=0.1,
        mole_fraction={'CO2':0.5, 'CO':0.5},
        isotope=1,
    )
s.plot('transmittance_noslit')
```

---

You need to modify the [docs/lbl/lbl.rst](https://github.com/radis/radis/blob/develop/docs/lbl/lbl.rst#calculating-spectra) file. That can be done online directly. 

The different possible syntaxes given in test_calc.py could be added there too.

File: docs/lbl/lbl.rst [hunk 1]
```diff
@@ -82,6 +82,107 @@ or supply an energy level database. In the latter case, you need to edit the
...
+    # Give isotope only
+    s = calc_spectrum(
+        wavelength_min=4165,
+        wavelength_max=5000,
+        Tgas=1000,
+        path_length=0.1,
+        isotope={"CO2": "1,2", "CO": "1,2,3"},
+        verbose=verbose,
+    )
+    assert set(s.conditions["molecule"]) == set(["CO2", "CO"])
```
Comment (erwanp) [inline docs/lbl/lbl.rst:148]:
```suggestion
```

Comment (ulyssed) [reply docs/lbl/lbl.rst]: Corrected!

**Comment by erwanp** [comment]
@jayanthchandra  do you want to review (as you started for #74 ) ? 
@CorentinGrimaldi as you'll be one of the first users?

**Comment by adityabharadwaj198** [comment]
@erwanp The syntax is clear enough, thanks for asking!

**Pull request merged**